src/Opcodes.hs

```diff
@@ -14,7 +15,54 @@ data Opcode = And
             | Sll
             | Srl
             | Sra
+            | Ldc
+            | Lw
+            | Sw
+            | ThreadFinished
     deriving (Show, Eq)
 
 mnemonic :: Opcode -> String
 mnemonic = map toLower . show
+
+hexOpcode :: Opcode -> Int
+hexOpcode opcode =
+  case opcode of
```

Comment:
can change this to
code `shiftL` 26
where code = case opcode of, so we only need one shift!